src/python/pants/backend/jvm/tasks/ivy_resolve.py

```diff
@@ -116,12 +116,24 @@ def execute(self):
                 self.context.log.info("Not generating a report. No resolution performed.")
             return
 
+        confs = self.get_options().confs
+        if "sources" not in confs and self.context.products.is_required_data(
+            "resolve_sources_signal"
+        ):
+            confs.append("sources")
+
+        if "javadoc" not in confs and self.context.products.is_required_data(
+            "resolve_javadocs_signal"
+        ):
+            confs.append("javadoc")
+
         executor = self.create_java_executor()
         results = self.resolve(
             executor=executor,
             targets=targets,
             classpath_products=compile_classpath,
-            confs=self.get_options().confs,
+            confs=confs,
+            invalidate_dependents=True,
```

Comment:
Is `invalidate_dependents=True` intended? as the default was False before.

Reply:
Thanks for catching that, I was using it to debug something.